extensions/rating-webchat-plugin/src/nodes/rating.ts

```diff
@@ -0,0 +1,160 @@
+import { createNodeDescriptor, INodeFunctionBaseParams } from "@cognigy/extension-tools";
+
+export interface IRatingCardParams extends INodeFunctionBaseParams {
+  config: {
+    initialRating: number;
+    title: string;
+    maxRatingValueHeart: number;
+    maxRatingValueStar: number;
+    precisionHeart: number;
+    precisionStar: number;
+    variant: string;
+    size: string;
+    rateButtonText: string;
+    label: string;
+    payload: string;
+  };
+}
+
+export const rating = createNodeDescriptor({
+  type: "ratingCard",
+  defaultLabel: "Show Rating Card",
+  fields: [
+    {
+      key: "title",
+      type: "cognigyText",
+      label: "Rating card title",
+      defaultValue: "Please give us your rating",
+      params: { required: true }
+    },
+    {
+      key: "variant",
+      type: "select",
+      label: "Variant",
+      defaultValue: "star",
+      params: {
+        options: [
+          { value: "star", label: "Star" },
+          { value: "heart", label: "Heart" },
+          { value: "emoji", label: "Emoji" }
+        ]
+      }
+    },
+    {
+      key: "initialRating",
+      label: "Initial rating when card shows up",
+      type: "slider",
+      params: { min: 0, max: 10, step: 1 }
+    },
+    { // TODO: remove these duplicates when advanced conditions are available
+      key: "maxRatingValueStar",
+      label: "Maximal Rating",
+      type: "slider",
+      defaultValue: 5,
+      params: { min: 2, max: 10, step: 1 },
+      condition: { key: "variant", value: "star" }
+    },
+    { // TODO: remove these duplicates when advanced conditions are available
+      key: "maxRatingValueHeart",
+      label: "Maximal Rating",
+      type: "slider",
+      defaultValue: 5,
+      params: { min: 2, max: 10, step: 1 },
+      condition: { key: "variant", value: "heart" }
+    },
+    { // TODO: remove these duplicates when advanced conditions are available
+      key: "precisionStar",
+      label: "Rating precision",
+      type: "slider",
+      defaultValue: "1",
+      params: { min: 0.5, max: 1, step: 0.5 },
+      condition: { key: "variant", value: "star" }
+    },
+    { // TODO: remove these duplicates when advanced conditions are available
+      key: "precisionHeart",
+      label: "Rating precision",
+      type: "slider",
+      defaultValue: "1",
+      params: { min: 0.5, max: 1, step: 0.5 },
+      condition: { key: "variant", value: "heart" }
+    },
+    {
+      key: "size",
+      label: "Size of rating icons",
+      type: "select",
+      defaultValue: "medium",
+      params: {
+        options: [
+          { value: "small", label: "Small" },
+          { value: "medium", label: "Medium" },
+          { value: "large", label: "Large" }
+        ]
+      }
+    },
+    {
+      key: "rateButtonText",
+      label: "Text on button that submits rating",
```

Comment:
The label should be short. Use the `description` field in order to write a sentence